storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -260,14 +262,18 @@ private void updateChainHead(
               commonAncestorSlot);
         }
 
+        final UInt64 epoch = compute_epoch_at_slot(newChainHead.getSlot());
+        final boolean epochTransition =
+            newChainHead.getSlot().equals(compute_start_slot_at_epoch(epoch));
```

Comment:
I don't think this is right.  We might wind up skipping multiple slots and so miss the first slot of the epoch.  I think it should check if the forkChoiceSlot from previousChainHead is from an earlier epoch than the forkChoiceSlot from newChainHead.

Reply:
hrm that's the other place i should be using the other function - then i dont need most of it, just pass into the function `is_start_slot_of_epoch`

Reply:
The point is that you're not looking for the first slot in the epoch.  You might be going from slot 30 to 50 in one update and that's an epoch transition even though you never got to slot 32.